Move CanBeExposed helper from factory_client_access

juanvallejo opened the pull request:
**Release note**:
```release-note
NONE
```

Moves additional helpers from the "factory_client_access". Continues work from https://github.com/kubernetes/kubernetes/pull/63984 and https://github.com/kubernetes/kubernetes/pull/63884

Moves CanBeExposed from factory_client_access.

cc @soltysh

juanvallejo committed:
move more CanBeExposed from factory_client_access

deads2k commented on pkg/kubectl/polymorphichelpers/helpers.go:
```diff
@@ -233,3 +276,162 @@ func SelectorsForObject(object runtime.Object) (namespace string, selector label
 }
+
+// Generators returns the generators for the provided command
+func Generators(cmdName string) map[string]kubectl.Generator {
```
No, these shouldn't live here.  Generators have been a disaster, but this isn't how to fix them.

deads2k commented on pkg/kubectl/cmd/expose.go:
```diff
@@ -358,3 +363,15 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 }
+
+// Check whether the kind of resources could be exposed
+func canBeExposed(kind schema.GroupKind) error {
```
This one is a polymorphic helper.

juanvallejo replied:
Done

juanvallejo committed:
move Pauser and Resumer from the factory

deads2k commented:
bazel needs updating

/lgtm

juanvallejo commented:
/test pull-kubernetes-e2e-kops-aws

juanvallejo commented:
/test pull-kubernetes-e2e-kops-aws

soltysh reviewed (approved):
/lgtm

juanvallejo committed:
move rollbacker from the factory

juanvallejo commented:
Rebased against latest master.
Added commits from https://github.com/kubernetes/kubernetes/pull/64121
Added commits from https://github.com/kubernetes/kubernetes/pull/64162

deads2k commented:
/retest

deads2k commented:
/retest
